### ISSUE TITLE
Support for High DPI Displays

### ISSUE THREAD

**Issue by nuttyartist** [nuttyartist/notes#39]
I got this email from someone:

> Hi guys,
> 
> Loving notes. I was wondering whether you were gonna support scaling for 4k? At the moment on my Dell XPS 15 (4k) the interface is tiny :-(

The best way to support High DPI Displays is to follow these guidelines:

- Always use the qreal versions of the QPainter drawing API.
- Size windows and dialogs in relation to the screen size.
- Replace hard-coded sizes in layouts and drawing code by values calculated from font metrics or screen size.

There are quick things we can do like setting some environment variables and attributes. But we should really aspire to quality support are users with High DPI Displays.

More info can be found here: http://doc.qt.io/qt-5/highdpi.html

**Comment by alex-spataru** [comment]
As you already mentioned, Qt 5.6 onwards supports high DPI screens. However, not everyone will have Qt 5.6 installed (especially Linux users). Here is the code that I use to obtain the scale factor of the screen:
```cpp
    /* Avoid UI scaling issues with Qt 5.6 */
#if QT_VERSION >= QT_VERSION_CHECK (5, 6, 0)
#if defined Q_OS_MAC
    QApplication::setAttribute (Qt::AA_EnableHighDpiScaling);
#else
    QApplication::setAttribute (Qt::AA_DisableHighDpiScaling);
#endif
#endif

    /* Calculate the scale factor of the screen */
    qreal ratio = (app.primaryScreen()->physicalDotsPerInch() / 100) * 0.9;

    /* Scale factor is too small */
    if (ratio < 1.2)
        ratio = 1;

    /* Mac already scales things */
#if defined Q_OS_MAC
    ratio = 1;
#endif
```
This code will output _1_ for normal density screens. On higher-density screens it will output the most appropriate value. 

I hope that this can help you!

**Comment by nuttyartist** [comment]
@alex-spataru does Qt 5.6 onwards support high DPI screens out-of-the-box?
Does someone has high screen display to test?

**Comment by julientaq** [comment]
I got a linux mint cinnamon 18.2 running with a 3k display, and i'd be happy to test any update of the app.

**Comment by nuttyartist** [comment]
@julientaq That's great, we will contact you when we'll work on this for our future release. Thanks :+1:

**Comment by OliverLew** [comment]
I am not sure if it is the right place, currently the layout has issue when set `Xft.dpi` to larger values (than 96, 144 in screenshot) in Xresources on Linux:
![Screenshot_2021-03-15_23-54-06](https://user-images.githubusercontent.com/12032219/111182563-fd641180-85e9-11eb-9653-a19154825c1b.png)
If I set qt scaling globally with `QT_SCALE_FACTOR=2` it works well, while `Xft.dpi` only changes font size I think.

**Comment by bjorn** [comment]
@OliverLew Yeah, this issue was also reported at #164 when larger fonts are used on Windows. The widget or delegate used there should probably calculate its size depending on the font size, whereas it currently appears to have a fixed height.

**Comment by nuttyartist** [comment]
@bjorn Exactly! That (plus upgrading the icons to suit bigger resolutions) would be the main objective of this issue.

**Comment by Darthagnon** [comment]
Guess it's the same bug, but I've noticed a lot of cut-off or strangely placed icons, as well as a glitch in the secondary settings dropdown (on right) (primary settings dropdown on left is ok)

![Notes_bGu2FMTr0G](https://github.com/nuttyartist/notes/assets/20792748/8ee86584-a759-43dd-9d58-bee8d990c703)

**Comment by nuttyartist** [comment]
First of all, I see you're using the Qt5 version, is that intentional? If so, why?

Second, I'm curious if in Kanban mode (which is supoprted in Qt 6+) you'll also have scaling issues. So if you could report back I'd appreciate it.

Theird, I think I understand why the Glitch with the Edtior Settings happen (Although I'll need to investigate more). Can you please open a new issue for that?

The rest, is probably an issue with scaling. Can you test to see what happens if you change your scaling in your OS settings? People that had this problem told me it fixed the issue. But of course we still need to find a way to make it work right out of the box. That's why I'm curious if there's also a scaling issue in the Kanban view - sicne it's written in QML and we're planning to migrate our entire UI to QML
